botocore/utils.py

```diff
@@ -3223,96 +3220,18 @@ def get_encoding_from_headers(headers, default='ISO-8859-1'):
         return default
 
 
-def calculate_md5(body, **kwargs):
-    if isinstance(body, (bytes, bytearray)):
-        binary_md5 = _calculate_md5_from_bytes(body)
-    else:
-        binary_md5 = _calculate_md5_from_file(body)
-    return base64.b64encode(binary_md5).decode('ascii')
-
-
-def _calculate_md5_from_bytes(body_bytes):
-    md5 = get_md5(body_bytes)
-    return md5.digest()
-
-
-def _calculate_md5_from_file(fileobj):
-    start_position = fileobj.tell()
-    md5 = get_md5()
-    for chunk in iter(lambda: fileobj.read(1024 * 1024), b''):
-        md5.update(chunk)
-    fileobj.seek(start_position)
-    return md5.digest()
-
-
-def _is_s3express_request(params):
-    endpoint_properties = params.get('context', {}).get(
-        'endpoint_properties', {}
-    )
-    return endpoint_properties.get('backend') == 'S3Express'
-
-
 def _has_checksum_header(params):
     headers = params['headers']
-    # If a user provided Content-MD5 is present,
-    # don't try to compute a new one.
-    if 'Content-MD5' in headers:
-        return True
 
     # If a header matching the x-amz-checksum-* pattern is present, we
-    # assume a checksum has already been provided and an md5 is not needed
+    # assume a checksum has already been provided by the user.
     for header in headers:
         if CHECKSUM_HEADER_PATTERN.match(header):
             return True
 
     return False
 
 
-def conditionally_calculate_checksum(params, **kwargs):
```

Comment:
Removing these may be breaking, I don't know if we've generally deleted functions out of the `utils.py` package.